app/models/person.rb

```diff
@@ -212,22 +222,22 @@ def find_first_by_auth_conditions(warden_conditions)
       end
     end
 
+    def find_by_email_param(person_params)
+      find_by_email(person_params.dig('email_addresses_attributes', '0', 'address'))
+    end
+
     def build_from_oauth_signup(auth, person_params={})
-      email, given, family = self.parse_info_from_auth(auth)
-      return unless email
-      new new_signup_attrs(auth, person_params, email, given, family)
+      from_oauth_signup :new, auth, person_params
     end
 
-    def create_or_update_from_oauth_signup(auth, person_params={})
+    def create_from_oauth_signup(auth, person_params={})
+      from_oauth_signup :create, auth, person_params
+    end
+
+    def from_oauth_signup(constructor, auth, person_params={})
       email, given, family = parse_info_from_auth(auth)
       return unless email
-      if person = Person.find_by_email(email)
-        save_omniauth(EmailAddress.find_by(address: email),
-                      Identity.find_for_oauth(auth),
-                      person.tap{ |p| p.update(person_params)})
-      else
-        create new_signup_attrs(auth, person_params, email, given, family)
-      end
+      send constructor, signup_attrs(auth, person_params, email, given, family)
```

Comment:
I feel that using `send` here obscures the meaning of what is trying to be done more than it helps saving the two lines of code needed to added to `build_from_oauth_signup` and `create_from_oauth_signup`

Reply:
i disagree, but less strongly than in other cases. perhaps we could pass `Person.new` and `Person.create` as lambdas and then `call` them here?

i don't think it's worth repeating ourself in this logic to avoid "digging" two lines up into a caller method that very clearly passes either `new` or `create` to this helper... also, i changed this logic several times in two places, which is what led me to extract it into one place.